packages/core/test/node/com.spec.ts

```diff
@@ -159,6 +158,46 @@ describe('Communication', () => {
         expect(handleMessageStub).to.have.not.been.called;
     });
 
+    it('forwards to pending envs wait until resolve', async () => {
+        const middlemanHost = new BaseHost();
+        const aHost = new BaseHost();
+        const bHost = new BaseHost();
+
+        const mainCom = new Communication(middlemanHost, 'middle');
+        const bCom = new Communication(bHost, 'bEnv');
+        const aCom = new Communication(aHost, 'aEnv');
+
+        disposables.add(mainCom);
+        disposables.add(bCom);
+        disposables.add(aCom);
+
+        aCom.registerEnv('middle', middlemanHost);
```

Comment:
Either register the middle to b as well or remove it from a to be persistent and less confusing